OmgvaPOS/src/ReservationManagement/Repository/ReservationRepository.cs

```diff
@@ -0,0 +1,55 @@
+using Microsoft.EntityFrameworkCore;
+using OmgvaPOS.Database.Context;
+using OmgvaPOS.ReservationManagement.Models;
+
+namespace OmgvaPOS.ReservationManagement.Repository;
+
+public class ReservationRepository : IReservationRepository
+{
+    private readonly OmgvaDbContext _context;
+
+    public ReservationRepository(OmgvaDbContext context)
+    {
+        _context = context;
+    }
+
+    public IEnumerable<Reservation> GetAll()
+    {
+        return _context.Reservations
+            .Include(r => r.User)
+            .Include(r => r.Customer)
+            .ToList();
+    }
+
+    public Reservation? GetById(long id)
+    {
+        return _context.Reservations
+            .Include(r => r.User)
+            .Include(r => r.Customer)
+            .FirstOrDefault(r => r.Id == id);
+    }
+
+    public Reservation Create(Reservation reservation)
+    {
+        _context.Reservations.Add(reservation);
+        _context.SaveChanges();
+        return reservation;
+    }
+
+    public Reservation Update(Reservation reservation)
+    {
+        _context.Reservations.Update(reservation);
```

Comment:
Important to strongly consider what effect may this operation have on historic data.

Reply:
Added TODO